src/index.js

```diff
@@ -117,8 +123,8 @@ export const Middleware = {
 
   timestamp: next => msg => next(merge(msg, {
     timestamp: (typeof msg.timestamp === 'Date' ?
-                   (msg.timestamp.getTime() + '000000') :
-                   Date.now() + '000000')
+                   Number((msg.timestamp.getTime() + '000000')) :
```

Comment:
My concern is that one can't reclaim the date because it's outside some 'sane' range in JS.

But I guess it's possible to do:
<img width="320" alt="screen shot 2016-04-27 at 22 37 20" src="https://cloud.githubusercontent.com/assets/193115/14867405/a41f0494-0cc8-11e6-92a5-c7e87b843ce6.png">

I'll merge this then.